components/contracts/dataGen.ts

```diff
@@ -97,24 +97,34 @@ export function generateCompletionData(
     gameVersion: GameVersion,
     isParentLocation = false,
 ): CompletionData {
-    // TODO(v6/v7): fetch actual statistics from the user's profile.
-
     const subLocation = getSubLocationByName(subLocationId, gameVersion)
 
-    return {
-        Level: 20,
-        MaxLevel: 20,
-        XP: 0,
-        Completion: 1,
-        XpLeft: 6000,
-        Id: isParentLocation
-            ? subLocationId
-            : subLocation?.Properties?.ParentLocation,
-        SubLocationId: subLocation?.Id,
-        HideProgression: false,
-        IsLocationProgression: true,
-        Name: null,
+    const locationId = isParentLocation
+        ? subLocationId
+        : subLocation?.Properties?.ParentLocation
+
+    const completionData = controller.masteryService.getCompletionData(
+        locationId,
+        gameVersion,
+        userId,
+    )
+
+    if (!completionData) {
+        return <CompletionData>{
+            Level: 20,
+            MaxLevel: 20,
```

Comment:
See max level comment in the mastery data. Also, if this is default completion data, why is it max level, and if it's meant to be fully completed, why is XpLeft not 0?

Reply:
This is literally what it was originally (just check the v5 branch). But you are correct, it doesn't make sense.

Reply:
The reason v5 does this is because it doesn't have a progression system so you can't get mastery, it used to be Level 1, but it was changed to allow easy comparison with official server responses when developing the challenge system.